requests-for-comments/2021-10-05-group-analytics.md

```diff
@@ -0,0 +1,214 @@
+# Request for Comments: *topic of the proposal*
+
+- **Created:** 2021-10-05
+- **Owner:** @macobo
+- **GitHub issues:** https://github.com/PostHog/posthog/issues/2247
+
+> While commenting on this, please comment on specific lines if possible, to thread the discussion.
+
+## Overview
+
+This RFC proposes a solution which would allow PostHog users to leverage insights and feature flags not only by users but also by groups (e.g. organizations, projects).
+
+## Background
+B2B products don’t deal directly with users, but with other companies and hence have more complicated needs around what kinds of metrics are most important for them.
+
+
+Lack of this feature has been a major driver for users wanting data exports in different analytics products, forcing users to roll their own analytics on top of data warehouses to get at their key metrics.
+
+### Example usecases
+
+
+- Show me how many active “organizations” there are
+- Show only pageviews from organizations who are paying us (filtering by group properties)
+- Have an overview of groups and their relations with users
+- Feature flags: roll out feature group-by-group
+
+For PostHog the product, the 3 group types we would use are:
+
+- Instances (e.g. cloud, posthog.someproduct.org, …)
+- Organizations
+- Projects
+
+## Semantics & Constraints
+
+### 1. Users can belong to multiple groups
+
+Some examples:
+
+1. Contractor type work (e.g. for farm management agronomists work for multiple farms at once)
+2. Billing happens on a different level (e.g. our projects vs organization)
+3. Users clients are global companies with a lot of local “branches” where execs need access to everything
+4. Internal users have access to debug issues on multiple accounts
+
+### 2. User is active in only “group” at any point in time
```

Comment:
worth specifying this is one group per "group type"?

e.g. I _can_ be in `org: PostHog` and `project: HogFlix` at the same time

Reply:
Added a clarifying note.

Reply:
Added a clarificaiton